File: 프로그래머스/0/181870. ad 제거하기/ad 제거하기.java
import java.util.*;

class Solution {
    public String[] solution(String[] strArr) {
        List<String> answer = new ArrayList<>();
        
        for(int i=0;i<strArr.length;i++){
            if(!strArr[i].contains("ad")){
               answer.add(strArr[i]);
            }
        }
        
        int len = answer.size();
        String[] sum = answer.toArray(new String[len]);

        return sum;
    }
}

File: 프로그래머스/0/181870. ad 제거하기/ad 제거하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char** solution(const char* strArr[], size_t strArr_len) {
    char** answer = (char**)malloc(strArr_len * 20 * sizeof(char));
    int len = 0;
    
    for(int i=0;i<strArr_len;i++){
        answer[i] = (char*)malloc(20 * sizeof(char));
        if(strstr(strArr[i], "ad") == NULL) answer[len++] = strArr[i]; 
    }
    
    return answer;
}